src/app.js

```diff
@@ -2,11 +2,87 @@
  * @param {Object} position
  * @param {Number} position.x
  * @param {Number} position.y
- * 
+ *
  * @returns {Array<position>}
  */
-export function getNearestShops(position) {
-  // code
-  
-  return [];
+import axios from "axios";
+
+const api = "https://blue-bottle-api-test.herokuapp.com/v1"
+const apiHeaders = {'Accept': 'application/json'}
+
+export async function getNearestShops(position) {
+    try {
+        const token = await getTokens()
+        const coffeeShops = await getCoffeeShops(token)
+        const nearShops = await getNearbyShops(position, coffeeShops, 3)
+        console.log('NearbyShops: ', nearShops) // I left this for viewing purposes
+        return nearShops
+
+    } catch (error) {
+        console.log(`ERROR ${error.type}: ${error.data}`)
+    }
 }
+
+function getTokens() {
+    return new Promise((resolve, reject) => {
+        axios.post(`${api}/tokens`, {headers: apiHeaders})
+            .then(response => {
+                resolve(response.data.token)
+            })
+            .catch(error => {
+                reject({type: 'TOKEN CREATE', data: error})
+            })
+    })
+}
+
+function getCoffeeShops(token) {
+    return new Promise((resolve, reject) => {
+        axios.get(`${api}/coffee_shops?token=${token}`, {headers: apiHeaders})
+            .then(response => {
+                resolve(response.data)
+            })
+            .catch(error => {
+                reject({type: 'COFFEESHOP FETCH', data: error})
+            })
+    })
+}
+
+function getNearbyShops(userPosition, shops, limit) {
+    return new Promise((resolve, reject) => {
+        let shopsWithUserDistance = []
+        for (let i = 0; i < shops.length; i++) {
+            shops[i]['distance'] = calculateDistance(shops[i]['x'], shops[i]['y'], userPosition['lat'], userPosition['lng']).toFixed(4)
+            shopsWithUserDistance.push(shops[i])
+        }
```

Comment:
This is hard to read, `calculateDistance` takes 4 arguments which by some standards is too much and it hints at the possibility for improvement. How would you reduce the number of parameters here?

And the way the parameters were passed is very verbose, it takes a lot of space horizontally, we use indexing to access each shop in the array, and we also use indexing to access properties, why not do `shops[i].x`?

Could you find some ways to make this easier on the eyes?